astropy/timeseries/downsample.py

```diff
@@ -167,14 +171,18 @@ def aggregate_downsample(time_series, *, time_bin_size=None, time_bin_start=None
     # Figure out which bin each row falls in by sorting with respect
     # to the bin end times
     indices = np.searchsorted(bin_end, ts_sorted.time[keep])
+
     # For time == bin_start[i+1] == bin_end[i], let bin_start takes precedence
-    if np.all(bin_start[1:] >= bin_end[:-1]):
+    if len(indices) and np.all(bin_start[1:] >= bin_end[:-1]):
         indices_start = np.searchsorted(ts_sorted.time[keep],
                                         np.minimum(bin_start, ts_sorted.time[-1]))
```

Comment:
```suggestion
        indices_start = np.searchsorted(subset.time, bin_start[bin_start <= ts_sorted.time[-1]])
```
Putting this forward now from https://github.com/astropy/astropy/pull/12527#issuecomment-984019182 as it fixes at least the known instances of Issue 3.